sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClient.cs

```diff
@@ -311,19 +314,20 @@ public virtual async Task<RecognizeCustomFormsOperation> StartRecognizeCustomFor
 
             SourcePath_internal sourcePath = new SourcePath_internal(formFileUri.ToString());
             ResponseWithHeaders<ServiceAnalyzeWithCustomModelHeaders> response = await ServiceClient.RestClient.AnalyzeWithCustomModelAsync(guid, includeTextDetails: recognizeOptions.IncludeTextContent, sourcePath, cancellationToken).ConfigureAwait(false);
-            return new RecognizeCustomFormsOperation(ServiceClient, modelId, response.Headers.OperationLocation);
+            return new RecognizeCustomFormsOperation(ServiceClient, Diagnostics, modelId, response.Headers.OperationLocation);
         }
 
         #endregion
 
         #region Training client
+
         /// <summary>
         /// Get an instance of a <see cref="FormTrainingClient"/> from <see cref="FormRecognizerClient"/>.
         /// </summary>
         /// <returns>An instance of a <see cref="FormTrainingClient"/>.</returns>
         public virtual FormTrainingClient GetFormTrainingClient()
         {
-            return new FormTrainingClient(_endpoint, _credential, _options?.Clone() ?? new FormRecognizerClientOptions());
+            return new FormTrainingClient(_endpoint, _credential, _options);
```

Comment:
Moved the `Clone` call to the constructor so users cannot modify the object once the client is created.